Objective-C/CBLArrayIndexConfiguration.m

```diff
@@ -0,0 +1,62 @@
+//
+//  CBLArrayIndexConfiguration.m
+//  CouchbaseLite
+//
+//  Copyright (c) 2024 Couchbase, Inc All rights reserved.
+//
+//  Licensed under the Apache License, Version 2.0 (the "License");
+//  you may not use this file except in compliance with the License.
+//  You may obtain a copy of the License at
+//
+//  http://www.apache.org/licenses/LICENSE-2.0
+//
+//  Unless required by applicable law or agreed to in writing, software
+//  distributed under the License is distributed on an "AS IS" BASIS,
+//  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//  See the License for the specific language governing permissions and
+//  limitations under the License.
+//
+
+#import "CBLArrayIndexConfiguration.h"
+#import "CBLIndexConfiguration+Internal.h"
+
+@implementation CBLArrayIndexConfiguration
+
+@synthesize path=_path, expressions=_expressions;
+
+- (instancetype) initWithPath: (NSString*) path
+                  expressions: (nullable NSArray<NSString*>*)expressions {
+    
+    if(!expressions){
+        self = [super initWithIndexType: kC4ArrayIndex
+                            expressions: @[@""]];
+    } else {
+        for (NSString* expression in expressions) {
+                if (!expression) {
```

Comment:
* NSArray/NSMutableArray doesn't allow nil so there is no need to check this for iOS.
* The only thing that is needed to evaluate is whether expressions is empty or not.